lambda/src/main/scala/pricemigrationengine/handlers/SalesforceNotificationDateUpdateHandler.scala

```diff
@@ -0,0 +1,104 @@
+package pricemigrationengine.handlers
+
+import java.time.{LocalDate, ZoneOffset}
+
+import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
+import pricemigrationengine.model.CohortTableFilter.{EmailSendComplete, EmailSendDateWrittenToSalesforce}
+import pricemigrationengine.model._
+import pricemigrationengine.services._
+import zio.clock.Clock
+import zio.console.Console
+import zio.{App, ExitCode, IO, Runtime, ZEnv, ZIO, ZLayer, clock}
+
+object SalesforceNotificationDateUpdateHandler extends App with RequestHandler[Unit, Unit] {
+
+  val main: ZIO[Logging with CohortTable with SalesforceClient with Clock, Failure, Unit] =
+    for {
+      cohortItems <- CohortTable.fetch(EmailSendComplete, None)
+      _ <- cohortItems.foreach(updateDateLetterSentInSF)
+    } yield ()
+
+  private def updateDateLetterSentInSF(
+      item: CohortItem
+  ): ZIO[Logging with CohortTable with SalesforceClient with Clock, Failure, Unit] =
+    for {
+      _ <- updateSalesforce(item)
+        .tapBoth(
+          e => Logging.error(s"Failed to write create Price_Rise in salesforce: $e"),
+          result => Logging.info(s"SalesforcePriceRise result: $result")
+        )
+      time <- clock.currentDateTime
+        .mapError { error =>
+          SalesforcePriceRiseCreationFailure(s"Failed to get currentTime: $error")
```

Comment:
Perhaps should be renamed as `SalesforcePriceRiseWriteFailure` now that its use is being extended